driver/t5layernorm_driver.hpp

```diff
@@ -0,0 +1,650 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef GUARD_MIOPEN_T5LAYERNORM_DRIVER_HPP
+#define GUARD_MIOPEN_T5LAYERNORM_DRIVER_HPP
+
+#include <../test/tensor_holder.hpp>
+#include <../test/verify.hpp>
+#include "InputFlags.hpp"
+#include "driver.hpp"
+#include "random.hpp"
+#include "tensor_driver.hpp"
+#include "timer.hpp"
+#include <algorithm>
+#include <cfloat>
+#include <cstdlib>
+#include <memory>
+#include <miopen/tensor.hpp>
+#include <numeric>
+#include <vector>
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloT5LayerNormForwardRunHost(miopenTensorDescriptor_t xDesc,
+                                     Tgpu* x,
+                                     Tgpu* weight,
+                                     Tcheck* yhost,
+                                     Tcheck* rstdhost,
+                                     float eps,
+                                     miopenNormMode_t mode)
+{
+    auto dims         = miopen::deref(xDesc).GetLengths();
+    size_t outer_size = 1;
+    size_t inner_size = dims[dims.size() - 1];
+
```

Comment:
I don't know if it's crucial here, but we may want to check inner_size because of the possible div0 on line 73:
```suggestion
MIOPEN_THROW_IF(inner_size == 0, "final dimension must be nonzero");
```

Reply:
Added to check where input is received.